bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/Proposal.scala

```diff
@@ -50,6 +53,14 @@ object Proposal {
 
   }
 
+  // find the target of opportunity of the proposal if possible
+  def toO(p: Option[Proposal]): Option[TooOption] = p.map(proposalClass.get) match {
+    case Some(q: QueueProposalClass)         => q.tooOption.some
+    case Some(l: LargeProgramClass)          => l.tooOption.some
+    case Some(f: FastTurnaroundProgramClass) => f.tooOption.some
+    case _                                   => none
+  }
```

Comment:
Agreed, it was difficult to give it a good name, on the UI it is shown as TOO, perhaps just ToO is good enough, changing it is pretty straightforward as it is an alias for the mutable class `TooOption` which cannot be changed without altering the schema (unless xjb lets you trick it somehow)